### PR TITLE
NEXUS-8305: Explicit DB TX

### PR DESCRIPTION
**Pull request by cstamas**
Explicitly re-using same ODB, and driving TX beginning/end. Preliminary testing shows it works and there is NO need for isolation anymore (as there are no overlapping connections no transactions over single thread).

Issue
https://issues.sonatype.org/browse/NEXUS-8305

CI
http://bamboo.s/browse/NX3-OSSF319

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java [hunk 1]
```diff
@@ -101,7 +105,9 @@ public StorageTxImpl(final BlobTx blobTx,
     this.assetEntityAdapter = checkNotNull(assetEntityAdapter);
     this.hook = checkNotNull(hook);
 
-    db.begin(TXTYPE.OPTIMISTIC);
+    if (!userManagedDb) {
+      db.begin(TXTYPE.OPTIMISTIC);
```
Comment (mrprescott) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java]: As discussed on the phone, the transaction-control aspect of StorageTx needs to be made consistent.  The close() and rollback() methods are still dealing with transactions, so the constructor should be calling db.begin() regardless of the user-managed mode.

Comment (cstamas) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java]: Ack fixed in https://github.com/sonatype/nexus-oss/commit/08b853737084db45647dcf63f5e80219ff8b4813

**Comment by mrprescott** [comment]
+1

**Comment by cstamas** [comment]
Merged as d6a235909f260ec64ab01e98e57dfd797885d95b

**Pull request closed without merging**